datanode/sqlstore/ledger.go

```diff
@@ -160,6 +164,85 @@ func (ls *Ledger) Query(
 	return &res, pageInfo, nil
 }
 
+func (ls *Ledger) Export(
+	ctx context.Context,
+	partyID, assetID string,
+	dateRange entities.DateRange,
+) ([]byte, string, error) {
+	if partyID == "" {
+		return nil, "", ErrLedgerEntryExportForParty
+	}
+
+	if assetID == "" {
+		return nil, "", ErrLedgerEntryExportForAsset
+	}
+
+	filter := &entities.LedgerEntryFilter{
+		SenderAccountFilter: entities.AccountFilter{
+			AssetID:  entities.AssetID(assetID),
+			PartyIDs: []entities.PartyID{entities.PartyID(partyID)},
+		},
+	}
+
+	aggregatedEntries, _, err := ls.Query(ctx, filter, dateRange, entities.CursorPagination{})
+	if err != nil {
+		return nil, "", err
+	}
+
+	path := fmt.Sprintf("%s-%s-%s-%s", partyID, assetID, dateRange.Start.String(), dateRange.End.String())
+	file, err := os.Create(path)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to create filename: %s", err.Error())
+	}
+	defer file.Close()
+
+	w := csv.NewWriter(file)
+	err = w.Write(
+		[]string{"VegaTime", "AssetId", "SenderPartyId", "SenderMarketId", "SenderAccountType", "ReceiverPartyId", "ReceiverMarketId", "ReceiverAccountType", "Quantity", "TransferType"},
+	)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to write headers to filename: %s", path)
+	}
+	defer w.Flush()
+
+	// Using Write
+	for _, record := range *aggregatedEntries {
+		fi, err := file.Stat()
+		if err != nil {
+			return nil, "", err
+		}
+
+		if fi.Size() >= 3670016 {
```

Comment:
I didn't meant about a hardcode size like this. we don't know with what max size a client is configured with. But I would have said that we could have some kind of pagination for it as we do for other endpoint. let's say that you are allowed to at most 1 month at a time of transfers, and the front end could request csv by paginating over that.

Reply:
Phil helped me figure out how to use the pagination - its done like this now.
Thanks!